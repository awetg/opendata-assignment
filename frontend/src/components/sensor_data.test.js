import React from "react";
import { create } from "react-test-renderer";
import SensorData from "./senso_data";

it("No value text is render for sensor with null values", () => {
  const tree = create(<SensorData data={{ name: "sensor1", value: null }} />);

  expect(tree).toMatchSnapshot();
});

it("Correct sensor name is render", () => {
  const tree = create(<SensorData data={{ name: "sensor4", value: 10.1 }} />);

  expect(tree).toMatchSnapshot();
});
