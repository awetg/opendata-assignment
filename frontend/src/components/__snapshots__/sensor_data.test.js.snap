// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct sensor name is render 1`] = `
<div
  className="card"
>
  <p
    className="sensor_label"
  >
    sensor4
  </p>
  <p
    className="sensor_value"
  >
    10.10
  </p>
</div>
`;

exports[`No value text is render for sensor with null values 1`] = `
<div
  className="card"
>
  <p
    className="sensor_label"
  >
    sensor1
  </p>
  <p
    className="sensor_value danger"
  >
    No Value
  </p>
</div>
`;
